### PR TITLE
Support more O3DS versions & change carry value hack

### PR DESCRIPTION
**Pull request by d3m3vilurr**
* Add all O3DS points
* Change carry bit hack

    In my opinion, #14 still can break in some cases
    And we can change increasing values at next time, and probably improve
    success rate.

### CODE REVIEW

File: source/timer.c [hunk 1]
```diff
@@ -28,26 +28,30 @@ static bool set_timer_negative(Handle timer, u32 kernel_callback_int, int error)
...
-  kernel_callback_shifted_goal = ((u64)0x80000000 + error) << 32;
+  kernel_callback_shifted_goal = ((u64)0x80000000) << 32;
   timeout = ((u64)(kernel_callback_int - 0x80000000) << 32);
+  switch (carry) {
+    case 1:
+      timeout += 0x55555556;
+      break;
+    case 2:
+      timeout += 0xaaaaaaab;
+      break;
```
Comment (d3m3vilurr) [inline source/timer.c:39]: these value is actually not magic.
64 bit divided values is `0x.....55555555` or `0x.....aaaaaaaa`
so i increased just last one.
but if we can catch right timing, can chage like `0x....55560000` :)

**Comment by d3m3vilurr** [comment]
well.. maybe i was misreaded. :/ it looks same... hum...

**Comment by nedwill** [comment]
Yeah this crazy behavior with the division in different cases affected me a lot. It'll be nice if we can work out any issues there.

**Comment by nedwill** [comment]
@d3m3vilurr https://github.com/nedwill/fasthax/blob/master/source/exploit.c#L133
I could never set the value to be negative before without this line. I think we need to actually figure out what is the best value to put there and figure it out dynamically.

**Comment by nedwill** [comment]
If you want to test some of those values, that would be nice. Right now I'm working on installing the real backdoor.

**Comment by d3m3vilurr** [comment]
i not thought need a9lh my o3ds before, but now changed mine and i'm installing this.
if you share values, i'll test them. :)

**Comment by nedwill** [comment]
@d3m3vilurr Can you add a constant for free kernel code space? `'\xff' * 40`, to copy in the backdoor implementation since it is removed now, like this: https://github.com/AuroraWright/Luma3DS/blob/master/source/patches.c#L243

**Comment by d3m3vilurr** [comment]
yeah i thought before, but i'm new in the scene, so also i need figuring for this.

**Comment by nedwill** [comment]
Oh, and it should be aligned to 4 bytes so the new instructions are valid. And you're doing great for being new!

**Comment by nedwill** [comment]
Hmm, actually I may just use the svcMyBackdoor. I really like this trick and no one should call that function since it's stubbed anyways. But we'll need to copy in the backdoor either way.

File: source/timer.c (hunk 1 above)
Comment (lotsa-spaghetti) [reply source/timer.c:39]: In which cases would my method not work? Or is that method just more efficient?

**Comment by ShinichiKudoSan** [comment]
Hey i've tested now the new build, made with devkitpro, copied into my sd card and started (old 3ds, 11.2.0-35E), we won! but then when i press start the console freeze and i have a redscreen. How can I help? Thanks guys ^:)

EDIT: Okey i've found out that when i press Start that there comes k11_exploit failed. First option + other 2 - ..

**Comment by grifnelle** [comment]
@ShinichiKudoSan can you drop a link to the file, I have devkitpro but its always giving me error 2

**Comment by ShinichiKudoSan** [comment]
Hey grifnelle here you go. :) I'm think you're devkitpro arent updated to last version
[fasthax-master.zip](https://github.com/nedwill/fasthax/files/681118/fasthax-master.zip)

**Comment by grifnelle** [comment]
@ShinichiKudoSan thank you kindly :]

**Comment by ShinichiKudoSan** [comment]
@grifnelle no problem :]
Okey now it has 2 versions on my o3ds. It attempts 0xf00 -> we won -> PRESS START -> shows!! k11 exploit fails. -> start crashes with redscreen (must power off the console)
Then now it attempts 0x2f00 -> goes we won -> PRESS START -> speedly hacks down and i cant see if failed or not + redscreen

**Comment by grifnelle** [comment]
mmm welp

**Comment by lotsa-spaghetti** [comment]
@d3m3vilurr does your method work better than mine?

**Comment by d3m3vilurr** [comment]
@kim-yannick i guess maybe, but I don't know really. I'm not checked success rate.
I just checked working with 9.2 o3ds.

**Comment by lotsa-spaghetti** [comment]
I think it will be the same, as the only random error that can occur is the sync with system clock, which happens rarely.

**Comment by grifnelle** [comment]
so then how do we get rid of the red screen crash

**Comment by nedwill** [comment]
@d3m3vilurr I'm going to merge early just to keep development moving. It's not fully working yet anyways so it's better if we keep working on the same branch.

**Pull request merged**